clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -108,54 +61,25 @@ public InputStream download(Artifact artifact) throws IOException {
     Path stagingPath =
         Paths.get(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString());
 
-    if (!isValidReference(repoReference)) {
-      throw new IOException(
-          "Artifact reference "
-              + repoReference
-              + " is invalid for artifact account with auth type "
-              + authType);
-    }
+    executor.clone(repoReference, remoteRef, stagingPath);
+    String repoBasename = getRepoBasename(repoReference);
+    Path outputFile = Paths.get(stagingPath.toString(), repoBasename + ".tgz");
 
-    try (Closeable ignored = () -> FileUtils.deleteDirectory(stagingPath.toFile())) {
-      log.info("Cloning git/repo {} into {}", repoReference, stagingPath.toString());
-      ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-      try (Git localRepository = clone(artifact, stagingPath, remoteRef)) {
-        log.info("Creating archive for git/repo {}", repoReference);
-        archiveToOutputStream(localRepository, outputStream, remoteRef, subPath);
-      } catch (GitAPIException e) {
-        throw new IOException(
-            "Failed to clone or archive git/repo " + repoReference + ": " + e.getMessage());
-      }
-      return new ByteArrayInputStream(outputStream.toByteArray());
-    }
-  }
+    log.info("Creating archive for git/repo {}", repoReference);
 
-  private Git clone(Artifact artifact, Path stagingPath, String remoteRef) throws GitAPIException {
-    // TODO(ethanfrogers): add support for clone history depth once jgit supports it
+    executor.archive(
+        Paths.get(stagingPath.toString(), repoBasename), remoteRef, subPath, outputFile);
 
-    return addAuthentication(Git.cloneRepository())
-        .setURI(artifact.getReference())
-        .setDirectory(stagingPath.toFile())
-        .setBranch(remoteRef)
-        .call();
+    return new FileInputStream(outputFile.toFile());
   }
 
-  private void archiveToOutputStream(
-      Git repository, OutputStream outputStream, String remoteRef, String subPath)
-      throws GitAPIException, IOException {
-
-    ArchiveCommand archiveCommand =
-        repository
-            .archive()
-            .setTree(repository.getRepository().resolve(remoteRef))
-            .setFormat("tgz")
-            .setOutputStream(outputStream);
-
-    if (!subPath.isEmpty()) {
-      archiveCommand.setPaths(subPath);
+  private String getRepoBasename(String url) {
+    Matcher matcher = GIT_URL_PATTERN.matcher(url);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException(
+          "Git repo url " + url + " doesn't match regex " + GIT_URL_PATTERN);
```

Comment:
IIRC, the previous implementation supported both SSH and HTTPS urls. HTTPS urls don't require the `.git` postfix so this seems like it'd be a backwards incompatible change.